### PR TITLE
wordy: update test to v1.5.0 (Fixes #1686 )

### LINKED ISSUE #1686: wordy: update tests and version file (v1.2.0 -> v1.5.0)
The tests for the [wordy exercise](https://github.com/exercism/java/tree/master/exercises/wordy/src/test/java) are not up-to-date with the [canonical data](https://github.com/exercism/problem-specifications/tree/master/exercises/wordy/canonical-data.json).
To check what has changed and if new tests have been added to the canonical data, please have a look at the [commit history](https://github.com/exercism/problem-specifications/commits/master/exercises/wordy/canonical-data.json) of the canonical data file for the exercise wordy.
Please update version and/or test file, and, if the new test(s) fail, fix the reference implementation.
Feel free to leave a comment if you have any questions. 

Background info: each exercise which has canonical data should have a version file. This file states which version of the canonical data the exercise implements. The version can be found at the top of the [canonical data file](https://github.com/exercism/problem-specifications/tree/master/exercises/wordy/canonical-data.json) for that exercise.

### PR DESCRIPTION
**Pull request by ThiagoSousaSantana**
wordy: update test to v1.5.0 (Fixes #1686 )

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by ThiagoSousaSantana** [comment]
The code failed in the first test case. But i think this test in the [canonical data](https://github.com/exercism/problem-specifications/tree/master/exercises/wordy/canonical-data.json) does not obey the structure of the "Problems" that are being passed in the other tests. What can i do to fix?

**Comment by lemoncurry** [comment]
Hei @ThiagoSousaSantana , I'm sorry to see that you closed your PR.

If you have troubles with fixing the implementation, we can always discuss the problems here in this PR. I would gladly help 😃 

Thank you again for your contributions.

**Comment by ThiagoSousaSantana** [comment]
Hi @lemoncurry, thanks for want help me 😃. I didn't understand how to fix this problem, i need to alter exercise solution?

**Comment by lemoncurry** [comment]
> Hi @lemoncurry, thanks for want help me 😃. I didn't understand how to fix this problem, i need to alter exercise solution?

Yes, exactly. The [reference implementation](https://github.com/exercism/java/blob/master/exercises/wordy/.meta/src/reference/java/WordProblemSolver.java) is no longer valid and needs changing since it is failing the new test.

In [this discussion](https://github.com/exercism/problem-specifications/pull/1382), @coriolinus proposes a simple change to the regex that might also work here.
Have you tried it?

File: exercises/wordy/src/test/java/WordProblemSolverTest.java [hunk 1]
```diff
@@ -112,12 +118,65 @@ public void testUnknownOperation() {
 
     @Ignore("Remove to run test")
     @Test
-    public void testInvalidQuestionFormat() {
+    public void testNomMathQuestion() {
         expectedException.expect(IllegalArgumentException.class);
```
Comment (lemoncurry) [inline exercises/wordy/src/test/java/WordProblemSolverTest.java]: There is a small typo, it should be "testNonMathQuestion".

File: exercises/wordy/src/test/java/WordProblemSolverTest.java [hunk 2]
```diff
@@ -112,12 +118,65 @@ public void testUnknownOperation() {
...
+    public void testNoOperandsOrOperators() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("I'm sorry, I don't understand the question!");
+
+        solver.solve("What is?");
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testTwoOperandsInARow() {
```
Comment (lemoncurry) [inline exercises/wordy/src/test/java/WordProblemSolverTest.java]: Also small typo, it should be "testTwoOperationsInARow"

File: exercises/wordy/src/test/java/WordProblemSolverTest.java (hunk 1 above)
Comment (ThiagoSousaSantana) [reply exercises/wordy/src/test/java/WordProblemSolverTest.java]: Sorry, I'm going to fix it now.

**Pull request merged**